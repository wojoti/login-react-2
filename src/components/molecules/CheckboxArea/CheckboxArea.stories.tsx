/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta, ComponentStory } from "@storybook/react";

import CheckboxArea from "./CheckboxArea";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Molecules/CheckboxArea",
  component: CheckboxArea,
} as ComponentMeta<typeof CheckboxArea>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof CheckboxArea> = (args) => (
  <CheckboxArea {...args} />
);

export const Example = Template.bind({});
Example.args = { name: "remember", id: "remember", label: "Remember Me?" };
