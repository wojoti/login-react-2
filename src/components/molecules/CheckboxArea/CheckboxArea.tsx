import Checkbox from "@components/atoms/Checkbox/Checkbox";
import Label from "@components/atoms/Label/Label";
import { forwardRef } from "react";
import Wrapper from "./CheckboxArea.style";

export interface CheckboxAreaProps {
  name: string;
  id: string;
  label: string;
  testId?: string;
}

export type CheckboxAreaRef = HTMLInputElement;

const CheckboxArea = forwardRef<CheckboxAreaRef, CheckboxAreaProps>(
  ({ name, id, label, testId }, ref) => (
    <Wrapper data-testid={testId}>
      <Checkbox name={name} id={id} testId="checkboxarea-checkbox" ref={ref} />
      <Label testId="checkboxarea-label">{label}</Label>
    </Wrapper>
  )
);
CheckboxArea.displayName = "CheckboxArea";

export default CheckboxArea;
