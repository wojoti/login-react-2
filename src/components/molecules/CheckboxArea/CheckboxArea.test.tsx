import { render, screen, within } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import CheckboxArea, { CheckboxAreaProps } from "./CheckboxArea";

const props: CheckboxAreaProps = {
  name: "name checkboxarea",
  id: "test-checkboxarea",
  label: "label checkboxarea",
  testId: "test-checkboxarea-id",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<CheckboxArea {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render checkboxarea", () => {
  render(<CheckboxArea {...props} />);
  const checkboxareaElement = screen.getByTestId("test-checkboxarea-id");
  expect(checkboxareaElement).toBeInTheDocument();
});

test("should render checkboxarea with certain label text", () => {
  render(<CheckboxArea {...props} />);
  const checkboxareaElement = screen.getByTestId("test-checkboxarea-id");
  expect(checkboxareaElement).toBeInTheDocument();
  const checkboxareaLabelElement =
    within(checkboxareaElement).getByTestId("checkboxarea-label");
  expect(checkboxareaLabelElement).toBeInTheDocument();
  expect(checkboxareaLabelElement).toHaveTextContent("label checkboxarea");
});

test("should render checkboxarea with certain checkbox id", () => {
  render(<CheckboxArea {...props} />);
  const checkboxareaElement = screen.getByTestId("test-checkboxarea-id");
  expect(checkboxareaElement).toBeInTheDocument();
  const checkboxareaCheckboxElement = within(checkboxareaElement).getByTestId(
    "checkboxarea-checkbox"
  );
  expect(checkboxareaCheckboxElement).toBeInTheDocument();
  expect(checkboxareaCheckboxElement).toHaveAttribute(
    "name",
    "name checkboxarea"
  );
});
