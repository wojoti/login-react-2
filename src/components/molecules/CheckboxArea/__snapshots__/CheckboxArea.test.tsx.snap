// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-right: 10px;
  accent-color: #f43f5e;
  width: 1rem;
  height: 1rem;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: inherit;
  font-size: 14px;
}

.c0 {
  width: 100%;
}

<div
  className="c0"
  data-testid="test-checkboxarea-id"
>
  <input
    className="c1"
    data-testid="checkboxarea-checkbox"
    id="test-checkboxarea"
    name="name checkboxarea"
    type="checkbox"
  />
  <span
    className="c2"
    data-testid="checkboxarea-label"
  >
    label checkboxarea
  </span>
</div>
`;
